erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -239,8 +239,9 @@ def before_submit(self):
 		self.create_serial_no_batch_no()
 
 	def on_submit(self):
-		if not self.wip_warehouse:
-			frappe.throw(_("Work-in-Progress Warehouse is required before Submit"))
+		if not self.skip_transfer:
+			if not self.wip_warehouse:
+				frappe.throw(_("Work-in-Progress Warehouse is required before Submit"))
```

Comment:
```suggestion
		if not self.skip_transfer and not self.wip_warehouse:
			frappe.throw(_("Work-in-Progress Warehouse is required before Submit"))
```

Merge conditions instead of nested?